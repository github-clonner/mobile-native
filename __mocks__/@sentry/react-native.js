export const captureMessage = jest.fn();
export default {};
