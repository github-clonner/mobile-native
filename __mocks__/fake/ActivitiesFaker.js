
function load(count, container) {

  let activities = [...Array(count)].map((_, i) => {
    const code = 'activityguid' + i;
    return {
      attachment_guid: false,
      blurb: false,
      container_guid:code,
      custom_data:false,
      custom_type:false,
      rowKey: 'something' + i,
      description:"Congratulations! ",
      edited:"",
      guid:code,
      mature:false,
      time_created: "1522036284",
      ownerObj:{
        guid: "824853017709780997",
        type: "user",
        subtype: false,
        time_created: "1522036284",
        // getAvatarSource: () => {
        //   return {
        //     source:'http://thisisaurl'
        //   }
        // }
      },
      shouldBeBlured: jest.fn(),
      message:"Message",
      title:'TITLE',
      owner_guid:"824853017709780997",
      parent_guid:"838106762591510528",
      perma_url:false,
      thumbnail_src:false,
      type:"activity",
      wire_totals: {
        tokens: 1000000000000000000
      },
      containerObj: container
      // _list: {
      //   viewed: {
      //     viewed: new Map([["1019155171608096768",true]]),
      //     addViewed: () => {
      //       return;
      //     }
      //   }
      // },
      // getThumbSource: () => {
      //   return {
      //     source:'http://thisisaurl'
      //   }
      // }
    }
  });

  return {
    'activities': activities,
    'load-next': 'aaaaaa',
    'load-previous': 'aaaaaa'
  }

}

/**
 * comment service data faker
 * @param {integer} count
 */
export function activitiesServiceFaker() {
  return { load: load}
}

