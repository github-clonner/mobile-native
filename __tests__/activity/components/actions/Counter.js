import 'react-native';
import React from 'react';
import { Text, TouchableOpacity } from "react-native";

import { activitiesServiceFaker } from '../../../../__mocks__/fake/ActivitiesFaker';

import renderer from 'react-test-renderer';
import Counter from '../../../../src/newsfeed/activity/actions/Counter';

describe('Owner component', () => {

  let screen;
  beforeEach(() => {

    const navigation = { navigate: jest.fn() };
    screen = renderer.create(
      <Counter count={100} />
    );

    jest.runAllTimers();
  });

  it('renders correctly', async () => {

    expect(screen).toMatchSnapshot();
  });


  it('should have Text', async () => {
    expect(screen.root.findAllByType('Text')).toHaveLength(1);
  });

  it('should _navToChannel on press ', () => {
    let touchables = screen.root.findAllByType('Text');
    expect(touchables[0].props['children']).toBe(100);
  });

});
