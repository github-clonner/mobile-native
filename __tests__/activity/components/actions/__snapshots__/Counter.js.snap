// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Owner component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "column",
      },
    ]
  }
>
  <Text
    style={
      Array [
        undefined,
        undefined,
        Object {
          "fontSize": 15,
        },
      ]
    }
  >
    100
  </Text>
</View>
`;
